src/Microsoft.Azure.WebJobs.Host/Bindings/Runtime/Binder.cs

```diff
@@ -0,0 +1,172 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.Azure.WebJobs.Host.Protocols;
+
+namespace Microsoft.Azure.WebJobs.Host.Bindings.Runtime
+{
+    /// <summary>
+    /// Class providing functionality for dynamically binding to WebJobs SDK attributes at runtime.
+    /// </summary>
+    public class Binder : IBinder, IWatchable, IDisposable
+    {
+        private readonly IAttributeBindingSource _bindingSource;
+        private readonly IList<IValueBinder> _binders = new List<IValueBinder>();
+        private readonly RuntimeBindingWatcher _watcher = new RuntimeBindingWatcher();
+        private readonly CollectingDisposable _disposable = new CollectingDisposable();
+        private readonly Dictionary<string, object> _bindingData;
+        private bool _disposed;
+
+        /// <summary>
+        /// For testing only.
+        /// </summary>
+        public Binder()
+        {
+        }
+
+        /// <summary>
+        /// Constructs a new instance.
+        /// </summary>
+        /// <param name="bindingSource">The binding source.</param>
+        internal Binder(IAttributeBindingSource bindingSource)
+        {
+            _bindingSource = bindingSource;
+
+            _bindingData = new Dictionary<string, object>(StringComparer.OrdinalIgnoreCase);
+            if (bindingSource.AmbientBindingContext.BindingData != null)
+            {
+                foreach (var pair in bindingSource.AmbientBindingContext.BindingData)
+                {
+                    _bindingData[pair.Key] = pair.Value;
+                }
+            }
+        }
+
+        /// <summary>
+        /// Gets the binding data.
+        /// </summary>
+        public virtual Dictionary<string, object> BindingData
```

Comment:
This replaces the `AmbientBindingContext` that I added to IBinderEx. I made that public before mainly to provide access to the `AmbientBindingContext.BindingData`.